pkg/apb/provision_or_update.go

```diff
@@ -76,31 +78,39 @@ func provisionOrUpdate(method executionMethod,
 	)
 	if err != nil {
 		log.Errorf("Problem executing apb [%s] provision - err: %v ", executionContext.PodName, err)
-		return executionContext.PodName, nil, err
+		e.finishWithError(err)
+		return
 	}
 
 	if instance.Spec.Runtime >= 2 || !instance.Spec.Bindable {
 		log.Debugf("watching pod for serviceinstance %#v", instance.Spec)
-		err := watchPod(executionContext.PodName, executionContext.Namespace, k8scli.Client.CoreV1().Pods(executionContext.Namespace), stateUpdates)
+		err := watchPod(executionContext.PodName, executionContext.Namespace,
+			k8scli.Client.CoreV1().Pods(executionContext.Namespace), e.updateDescription)
 		if err != nil {
 			log.Errorf("Provision or Update action failed - %v", err)
-			return executionContext.PodName, nil, err
+			e.finishWithError(err)
+			return
 		}
 	}
 
 	if !instance.Spec.Bindable {
-		return executionContext.PodName, nil, nil
+		e.finishWithSuccess()
+		return
 	}
 
 	creds, err := ExtractCredentials(
 		executionContext.PodName,
 		executionContext.Namespace,
 		instance.Spec.Runtime,
 	)
+	e.extractedCredentials = creds
+
```

Comment:
same comment as bind, are we sure we want to set this in case of an err. If so, okay.